Userland/Libraries/LibCore/IOSurface.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2024, Aliaksandr Kalenik <kalenik.aliaksandr@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Forward.h>
+#include <AK/Noncopyable.h>
+#include <LibCore/MachPort.h>
+
+namespace Core {
+
+class IOSurfaceHandle {
+    AK_MAKE_NONCOPYABLE(IOSurfaceHandle);
+
+public:
+    IOSurfaceHandle(IOSurfaceHandle&& other);
+    IOSurfaceHandle& operator=(IOSurfaceHandle&& other);
+
+    static IOSurfaceHandle create(size_t width, size_t height);
+    static IOSurfaceHandle from_mach_port(MachPort port);
+
+    MachPort create_mach_port() const;
+
+    size_t width() const;
+    size_t height() const;
+    size_t bytes_per_element() const;
+    void* data() const;
+
+    virtual ~IOSurfaceHandle();
```

Comment:
What's this for?

Reply:
oops, `virtual` is not necessary here indeed.